### PR TITLE
RFC: Babel all the things

### PR DESCRIPTION
**Pull request by dmbch**
Apart from making sure we transpile everything (except polyfills), I have added a small middleware to determine if the current browser is supported according to our browsers config.

Relates to and possibly resolves #175

### CODE REVIEW

File: packages/webpack/lib/configs/develop.js [hunk 1]
```diff
@@ -108,7 +106,7 @@ module.exports = function getConfig(config) {
     },
     plugins: [
       new HotModuleReplacementPlugin(),
-      new NamedModulesPlugin(),
+      new HashedModuleIdsPlugin(),
```
Comment (robin-drexler) [inline packages/webpack/lib/configs/develop.js]: Wouldn't it be better to keep the `NamedModulesPlugin` here for easier debugging? 
Or is there a need to switch them?

File: packages/webpack/lib/middleware/agent.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const { matchesUA } = require('browserslist-useragent');
+
+module.exports = function createAgentMiddleware({ config: { browsers } }) {
```
Comment (robin-drexler) [inline packages/webpack/lib/middleware/agent.js]: I really like the idea and the feature but wonder if it makes more sense in hops as this seems a bit too far from the core tooling platform untool ought to be.

Also, a test would be highly appreciated.

**Comment by ZauberNerd** [comment]
So I did a couple of tests on some of our internal and external projects and got the following numbers:

```
cxxxs-frontend (internal, postcss):
- babel only esnext:
    - dev build
        - with cold cache: `hops build`: ~1m
        - with warm cache: `hops build`: ~40s
    - prod build
        - with cold cache: `hops build -p`: ~2m
        - with warm cache `hops build -p`: ~1m10s
    - watch
        - with cold cache: `hops develop`: ~50s
        - with warm cache `hops develop`: -40s
        - incremental build: ~5s
- babel everything:
    - dev build
        - with cold cache: `hops build`: ~2m
        - with warm cache: `hops build`: ~1m10s
    - prod build
        - with cold cache: `hops build -p`: ~3m
        - with warm cache `hops build -p`: ~1m30s
    - watch
        - with cold cache: `hops develop`: ~1m20s
        - with warm cache `hops develop`: ~40s
        - incremental build: ~5s

cxxxxt-frontend (internal, graphql, postcss):
- babel only esnext:
    - dev build
        - with cold cache: `hops build`: ~1m
        - with warm cache: `hops build`: ~40s
    - prod build
        - with cold cache: `hops build -p`: ~2m10s
        - with warm cache `hops build -p`: ~1m20s
    - watch
        - with cold cache: `hops develop`: ~50s
        - with warm cache `hops develop`: -30s
        - incremental build: ~6s
- babel everything:
    - dev build
        - with cold cache: `hops build`: ~2m10s
        - with warm cache: `hops build`: ~1m10s
    - prod build
        - with cold cache: `hops build -p`: ~3m30s
        - with warm cache `hops build -p`: ~1m50s
    - watch
        - with cold cache: `hops develop`: ~1m30s
        - with warm cache `hops develop`: ~40s
        - incremental build: ~7s

foosball-trainer (typescript, styled-components):
- babel only esnext:
    - dev build
        - with cold cache: `hops build`: ~25s
        - with warm cache: `hops build`: ~20s
    - prod build
        - with cold cache: `hops build -p`: ~40s
        - with warm cache `hops build -p`: ~20s
    - watch
        - with cold cache: `hops develop`: ~20s
        - with warm cache `hops develop`: -15s
        - incremental build: ~1s
- babel everything:
    - dev build
        - with cold cache: `hops build`: ~35s
        - with warm cache: `hops build`: ~20s
    - prod build
        - with cold cache: `hops build -p`: ~1m
        - with warm cache `hops build -p`: ~25s
    - watch
        - with cold cache: `hops develop`: ~30s
        - with warm cache `hops develop`: ~15s
        - incremental build: ~1s
```

**Comment by robin-drexler** [comment]
@dmbch do you have any clue why those react packages have such an impact on build times?

**Comment by dmbch** [comment]
@robin-drexler they are rather large - >100kB minified - and contain quite a lot of files. I guess that is what slows down transpilation: lots of file I/O.

**Comment by robin-drexler** [comment]
Regarding reproducing the issue. Have you tried to add the Symbol core-js polyfill imort right after the react import in your [index.js](https://github.com/untool/untool/blob/master/tests/fixtures/untest/index.js)?

This could be enough to trigger the issue. Otherwise I can try to reproduce on Monday.

**Pull request merged**